src/runtime/internal/dom.ts

```diff
@@ -132,7 +132,7 @@ export function append(target: NodeEx, node: NodeEx) {
 export function insert(target: NodeEx, node: NodeEx, anchor?: NodeEx) {
 	if (is_hydrating && !anchor) {
 		append(target, node);
-	} else if (node.parentNode !== target || (anchor && node.nextSibling !== anchor)) {
+	} else if (node.parentNode !== target || node.nextSibling !== anchor) {
```

Comment:
`node.parentNode !== target || (anchor && (node.nextSibling !== anchor)) || ((!anchor) && (node.nextSibling !== null))` might be more correct since `anchor` is `Node | undefined` and `nextSibling` is `Node | null`

Reply:
@hbirler 

Can I think of it like `node.nextSibling != anchor`?

Reply:
or `node.nextSibling !== (anchor || null)`

Reply:
Yes! I believe it would look much simpler with your suggestions.

Reply:
I applied it. Thank you.